src/mscorlib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2,12 +2,56 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
 
 namespace System.Runtime.Intrinsics
 {
     [Intrinsic]
+    [DebuggerDisplay("{DisplayString,nq}")]
+    [DebuggerTypeProxy(typeof(Vector128DebugView<>))]
     [StructLayout(LayoutKind.Sequential, Size = 16)]
-    public struct Vector128<T> where T : struct {}
+    public struct Vector128<T> where T : struct
+    {
+        // These fields exist purely so debug view works (https://github.com/dotnet/coreclr/issues/15694)
+        private byte _00;
```

Comment:
As per the comment, these exist so the debugger will properly marshal bytes. There are a couple bugs on GitHub tracking this and a couple internal ones as well.

Reply:
Do these `byte` fields affect how `Vector128<T>` fields are aligned?

Reply:
I don't think so. `System.Numerics.Vector<T>` also has byte fields (although public, and indirected through the `Register` struct).

Is there a good way of validating this today?

Reply:
Create struct like this and check the offsets of `_b` and `_d`:

```csharp
struct MyStruct {
    byte _a;
    Vector<long> _b;
    byte _c;
    Vector<double> _d;
}
```

It is possible that the alignment is not right already. I think the fields `_b` and `_d` should be aligned at 8 bytes in this case.

Reply:
Looks like the alignment is not right already.

I ran four tests (multiple, separate, process executions to ensure I got the correct results):
```C#
public struct MyStruct128 {
    public byte _a;
    public Vector128<long> _b;
    public byte _c;
    public Vector128<double> _d;
}

public static void Test128_1() {
    Vector128<ulong> val;

    Console.Write("Vector128 is ");
    TestAlignment(Unsafe.AsPointer(ref val));
}

public static void Test128_2() {
    byte _a;
    Vector128<ulong> val;

    Console.Write("Vector128 is ");
    TestAlignment(Unsafe.AsPointer(ref val));
}

public static void Test128_3() {
    MyStruct128 val;

    Console.Write("Vector128 is ");
    TestAlignment(Unsafe.AsPointer(ref val._b));
}

public static void Test128_4() {
    MyStruct128 val;

    Console.Write("Vector128 is ");
    TestAlignment(Unsafe.AsPointer(ref val._d));
}
```

I saw the following on x64:
* Test128_1: val is 16-byte aligned
* Test128_2: val is 16-byte aligned
* Test128_3: val._b is 1-byte aligned
* Test128_3: val._b is 2-byte aligned

I saw the following on x86:
* Test128_1: val is 4-byte aligned
* Test128_2: val is 8-byte aligned
* Test128_3: val._b is 1-byte aligned
* Test128_3: val._b is 2-byte aligned

Reply:
Does VM's struct layout have any special understanding of these types? While it's well known that they won't be 16/32 byte aligned due to the fact that GC only supports 8 byte alignment one would expect that they're aligned like any other struct that contains a field of type `T`. But since they don't actually contain any fields of type `T` the alignment can be as small as 1.

Reply:
It does not today. It can either be special cased in the type loader; or by changing the padding fields to `T` or `long`.

Reply:
It should probably be special cased in the type loader. The fields are currently a temporary solution to the bug in the debugger.

However, if we did go with keeping the fields (rather than special casing), I would vote we use `long` or `ulong`, as that would provide the most consistent experience (rather than ending up with one alignment for `Vector128<byte>` and another for `Vector128<float>`).

Reply:
Adding a padding field in CoreLib is less pain than special casing a generic type in the type loader.

> vote we use long or ulong

Sounds good to me.

Reply:
Fixed.

Reply:
Hmm, if we do special case the type loader then maybe we should actually make these types 16/32 bytes aligned, as far as struct layout is concerned. Otherwise you may end up with surprised people wondering why the default .NET struct layout doesn't match the C/C++ struct layout.

Reply:
That would be desirable to some. However, that might be worth a distinct issue for tracking/discussing.

Reply:
We don't currently have runtime support for 16/32 byte alignment. @jkotas can probably chime in on how difficult it would be to add.

Reply:
I think, based on the previous discussions, it would need to be only for stack-based allocations (modifying the GC to support arbitrary alignments is hard).

However, that might be confusing and might also be non-trivial to achieve.

Reply:
I was referring the layout of structs containing vectors. In C/C++ the following struct
```C
struct foo {
    char b;
    __m256i v;
};
```
is 64 byte in size, there's 31 bytes of padding between `b` and `v`.

As is now, the only way to reproduce such a layout in .NET is to use explicit struct layout.

The fact that such a struct is not guaranteed to be 32 byte aligned when placed in the GC heap is a separate issue.

Reply:
Are we sure all platforms will have the same layout requirements?

> the only way to reproduce such a layout in .NET is to use explicit struct layout.

Is that a bad thing?

Reply:
> We don't currently have runtime support for 16/32 byte alignment. @jkotas can probably chime in on how difficult it would be to add.

Non-trivial non-pay-for-play GC feature. Related discussion:

https://github.com/dotnet/coreclr/issues/5195
https://github.com/dotnet/corefx/issues/22790

Reply:
> Are we sure all platforms will have the same layout requirements?

Good question. I suppose you may know how this works on ARM? It's probably even more interesting if you drag in ARM's SVE.

> Is that a bad thing?

Hard to say. It may be problematic if it turns out that different platforms use different alignments. For example, if ARM's 32 byte vectors are only 16 byte aligned then my `foo` struct will have only 15 bytes of padding. Then you'll have to create different .NET structs for ARM and x86 because the explicit struct layout that you can define in .NET using attributes is practically hardcoded at compile time.